modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -3205,6 +3212,18 @@ private void resetSizeEstimates() {
         this.estimatedSize = 1d;
     }
 
+    private void setContentX(double x) {
+        double snappedX = snapPositionX(x);
+        clipView.setLayoutX(-snappedX);
+        clipRect.setLayoutX(snappedX);
+    }
+
+    private void setContentY(double y) {
+        double snappedY = snapPositionY(y);
+        clipView.setLayoutY(-snappedY);
+        clipRect.setLayoutY(snappedY);
+    }
+
```

Comment:
The method names sound a little confusing to me. The methods only set the values for clip rect and clip view.
The hbar.getValue() represents the x value for the visible content of control.
I would recommend to rename these methods to something as, setClipLayoutX or setClipX..

Reply:
I chose this name to be consistent with the `ScrollPaneSkin`, but I agree that this is not the best name. 
`setClipLayoutX` sounds good to me, will change.